redis/mutation_utils.cc

```diff
@@ -51,6 +52,29 @@ atomic_cell make_cell(const schema_ptr schema,
     return atomic_cell::make_live(type, api::new_timestamp(), value, atomic_cell::collection_member::no);
 }  
 
+// cas_request sub-class for write_strings
+class write_strings_cas_request : public service::cas_request {
+private:
+    service::storage_proxy& _proxy;
+    schema_ptr _schema;
+    bytes& _key;
+    bytes& _data;
+    long _ttl;
+
+public:
+    write_strings_cas_request(service::storage_proxy& proxy, redis::redis_options& options, bytes& key, bytes& data, long ttl)
+    : _proxy(proxy), _schema(get_schema(proxy, options.get_keyspace_name(), redis::STRINGs)), _key(key), _data(data), _ttl(ttl)  {}
+    virtual ~write_strings_cas_request() = default;
+    virtual std::optional<mutation> apply(foreign_ptr<lw_shared_ptr<query::result>> qr, const query::partition_slice& slice, api::timestamp_type ts) override {
+        const column_definition& column = *_schema->get_column_definition(redis::DATA_COLUMN_NAME);
+        auto pkey = partition_key::from_single_value(*_schema, _key);
+        auto m = mutation(_schema, std::move(pkey));
+        auto cell = make_cell(_schema, *(column.type.get()), _data, _ttl);
```

Comment:
You **must** use the given timestamp `ts` in your mutation!
If you don't understand why, I can explain.

Reply:
Applied `ts` to the mutation.